src/stratum/stratum.cpp

```diff
@@ -505,6 +516,16 @@ namespace merit
             return true;
         }
 
+        void Client::switch_pool()
+        {
+            if(reserve_pools.empty())
+                return;
+
+            reserve_pools.push_back(_url);
```

Comment:
change reserve_pools type to a std::deque and use pop_front()/push_back methods.

Reply:
On second thought, just keeping an index and round robinning it should be good enough.